subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/billing/PlayBillingManager.kt

```diff
@@ -108,9 +110,7 @@ class RealPlayBillingManager @Inject constructor(
     }
 
     private fun connectAsyncWithRetry() {
-        if (connectionJob?.isActive == true) return
-
-        connectionJob = coroutineScope.launch {
+        connectionJob += coroutineScope.launch(dispatcherProvider.io()) {
```

Comment:
I could not see a reason why we'd not cancel the previous job and re-launch a new one.

connectAsyncWithRetry is called from `onCreate`, this is when the app is first created and then `onBillingClientDisconnected`, in both cases I think it's just fine (probably better from stateless stand point) to just cancel and re-launch (?)